inst/include/odeproblem.h

```diff
@@ -38,7 +39,7 @@
 class odeproblem;
 
 //! vector of <code>datarecord</code> objects for one <code>ID</code>
-typedef std::vector<rec_ptr> reclist;
+typedef std::deque<rec_ptr> reclist;
```

Comment:
This is the main change. `reclist` should be used all over the code base. I found some places were I wasn't using that and I updated them to use `reclist`